// react-jsx-highcharts/dist/es/components/Highcharts3dChart/Highcharts3dChart.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import React, { Component } from 'react';
import HighchartsChart from '../HighchartsChart';
import Options3d from '../Options3d';

var Highcharts3dChart =
/*#__PURE__*/
function (_Component) {
  _inherits(Highcharts3dChart, _Component);

  function Highcharts3dChart() {
    _classCallCheck(this, Highcharts3dChart);

    return _possibleConstructorReturn(this, _getPrototypeOf(Highcharts3dChart).apply(this, arguments));
  }

  _createClass(Highcharts3dChart, [{
    key: "render",
    value: function render() {
      var _this$props = this.props,
          children = _this$props.children,
          alpha = _this$props.alpha,
          axisLabelPosition = _this$props.axisLabelPosition,
          beta = _this$props.beta,
          depth = _this$props.depth,
          fitToPlot = _this$props.fitToPlot,
          frame = _this$props.frame,
          viewDistance = _this$props.viewDistance,
          rest = _objectWithoutProperties(_this$props, ["children", "alpha", "axisLabelPosition", "beta", "depth", "fitToPlot", "frame", "viewDistance"]);

      var chart = {
        options3d: {
          enabled: true
        }
      };
      var zAxis = {
        id: 'zAxis'
      };
      return React.createElement(HighchartsChart, _extends({
        chart: chart,
        zAxis: zAxis
      }, rest), React.createElement(Options3d, {
        alpha: alpha,
        axisLabelPosition: axisLabelPosition,
        beta: beta,
        depth: depth,
        fitToPlot: fitToPlot,
        frame: frame,
        viewDistance: viewDistance
      }), children);
    }
  }]);

  return Highcharts3dChart;
}(Component);

export default Highcharts3dChart;
