// react-jsx-highcharts/dist/es/components/ZAxis/ZAxis.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Axis from '../Axis';

var ZAxis =
/*#__PURE__*/
function (_Component) {
  _inherits(ZAxis, _Component);

  function ZAxis() {
    _classCallCheck(this, ZAxis);

    return _possibleConstructorReturn(this, _getPrototypeOf(ZAxis).apply(this, arguments));
  }

  _createClass(ZAxis, [{
    key: "render",
    value: function render() {
      return React.createElement(Axis, _extends({}, this.props, {
        id: "zAxis",
        isX: false,
        dynamicAxis: false
      }));
    }
  }]);

  return ZAxis;
}(Component);

_defineProperty(ZAxis, "defaultProps", {
  type: 'linear'
});

ZAxis.Title = Axis.Title;
export default ZAxis;
