// react-jsx-highcharts/dist/es/components/Loading/Loading.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _assertThisInitialized from "@babel/runtime/helpers/esm/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import { Component } from 'react';
import PropTypes from 'prop-types';
import { attempt } from 'lodash-es';
import getModifiedProps from '../../utils/getModifiedProps';

var Loading =
/*#__PURE__*/
function (_Component) {
  _inherits(Loading, _Component);

  function Loading() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, Loading);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(Loading)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "updateLoading", function (config) {
      var chart = _this.props.getChart();

      chart.update({
        loading: config
      }, true);
    });

    return _this;
  }

  _createClass(Loading, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this$props = this.props,
          children = _this$props.children,
          isLoading = _this$props.isLoading,
          getChart = _this$props.getChart,
          rest = _objectWithoutProperties(_this$props, ["children", "isLoading", "getChart"]);

      this.updateLoading(rest);

      if (isLoading) {
        var chart = getChart();
        chart.showLoading(children);
      }
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this$props2 = this.props,
          children = _this$props2.children,
          isLoading = _this$props2.isLoading,
          getChart = _this$props2.getChart,
          rest = _objectWithoutProperties(_this$props2, ["children", "isLoading", "getChart"]);

      var modifiedProps = getModifiedProps(prevProps, rest);

      if (modifiedProps !== false) {
        this.updateLoading(modifiedProps);
      }

      if (isLoading !== prevProps.isLoading) {
        var chart = getChart();
        if (isLoading) chart.showLoading(children);
        if (!isLoading) chart.hideLoading();
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      var chart = this.props.getChart();
      attempt(chart.hideLoading);
    }
  }, {
    key: "render",
    value: function render() {
      return null;
    }
  }]);

  return Loading;
}(Component);

_defineProperty(Loading, "defaultProps", {
  isLoading: true
});

export default Loading;
