// react-jsx-highcharts/dist/es/components/Tooltip/Tooltip.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _assertThisInitialized from "@babel/runtime/helpers/esm/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { attempt } from 'lodash-es';
import getModifiedProps from '../../utils/getModifiedProps';

var Tooltip =
/*#__PURE__*/
function (_Component) {
  _inherits(Tooltip, _Component);

  function Tooltip() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, Tooltip);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(Tooltip)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "updateTooltip", function (config) {
      var chart = _this.props.getChart();

      chart.update({
        tooltip: config
      }, false);

      _this.props.needsRedraw();
    });

    return _this;
  }

  _createClass(Tooltip, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var _this$props = this.props,
          children = _this$props.children,
          getHighcharts = _this$props.getHighcharts,
          getChart = _this$props.getChart,
          needsRedraw = _this$props.needsRedraw,
          rest = _objectWithoutProperties(_this$props, ["children", "getHighcharts", "getChart", "needsRedraw"]);

      var Highcharts = getHighcharts();
      var chartObj = getChart().object;
      chartObj.tooltip = new Highcharts.Tooltip(chartObj, _objectSpread({}, Highcharts.defaultOptions && Highcharts.defaultOptions.tooltip, rest));
      this.updateTooltip(rest);
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var modifiedProps = getModifiedProps(prevProps, this.props);

      if (modifiedProps !== false) {
        this.updateTooltip(modifiedProps);
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      attempt(this.updateTooltip, {
        enabled: false
      });
    }
  }, {
    key: "render",
    value: function render() {
      return null;
    }
  }]);

  return Tooltip;
}(Component);

_defineProperty(Tooltip, "defaultProps", {
  children: null,
  enabled: true
});

export default Tooltip;
